ra/ra.go

```diff
@@ -884,9 +884,18 @@ func (ra *RegistrationAuthorityImpl) failOrder(
 func (ra *RegistrationAuthorityImpl) FinalizeOrder(ctx context.Context, req *rapb.FinalizeOrderRequest) (*corepb.Order, error) {
 	order := req.Order
 
-	// Only pending orders can be finalized
-	if *order.Status != string(core.StatusPending) {
-		return nil, berrors.InternalServerError("Order's status (%q) was not pending", *order.Status)
+	// If the `OrderReadyStatus` feature flag is disabled, only pending orders can
+	// be finalized. This is the legacy ACME behaviour prior to the ready state
+	// being introduced in draft-10
+	expectedOrderStatus := string(core.StatusPending)
+	// If the `OrderReadyStatus` feature flag is enabled, only **ready** orders
```

Comment:
@jsha I agree with the deployability concern. I disagree that we should allow finalizing in either state and remove the flag entirely.

The flag & enforcing that an order is status "final" here helps prevent invaliding orders unnecessarily if a client POSTs finalization before the authzs are ready. In the future when we have the flag enabled in the RA/SA your order will be "pending" when your authzs aren't ready and we can reject the finalization immediately without transitioning the order status to invalid. If we allow finalization of "pending" then the finalization will fail due to missing authorizations and the order will become invalid.

Reply:
In this specific case I also think its harmless for the RA with the flag enabled to finalize an order through a SA that doesn't have the flag set. The SA's flag is only used for order retrieval.

Reply:
I can't think of a clever way to address the need for the WFE/RA/SA to get the feature flag in lockstep.

I'll adjust the PR this afternoon to accept both and remove the legacy "pending" acceptance code after the flag is enabled. This will likely be a 2-4 weeks out by the time we merge, deploy and announce with enough lead time.

Reply:
@jsha Addressed in 9b4b32e - Thanks!